hpx/parallel/util/loop.hpp

```diff
@@ -83,21 +83,71 @@ namespace hpx { namespace parallel { namespace util
                 return it;
             }
         };
+
+        template <typename Iterator>
+        struct loop<parallel::execution::parallel_unsequenced_policy, Iterator>
+        {
+            ///////////////////////////////////////////////////////////////////
+            template <typename Begin, typename End, typename F>
+            HPX_HOST_DEVICE HPX_FORCEINLINE
+            static Begin call(Begin it, End const end, F && f)
+            {
+#if defined(HPX_MSVC)
+#pragma loop(ivdep)
+#elif defined(HPX_GCC_VERSION)
+#pragma GCC ivdep
+#pragma omp simd
+#elif defined(HPX_INTEL_VERSION) || defined(HPX_CLANG_VERSION)
+#pragma ivdep
+#pragma omp simd
+#endif
```

Comment:
I'd like to leave omp out of the picture here since we'd require users to compile with `-fopenmp` then.

Reply:
What about other compilers, should we issue a warning?